src/events/neo4j.js

```diff
@@ -0,0 +1,265 @@
+// All Session and Transaction properties are private
+/* eslint-disable no-underscore-dangle, prefer-rest-params */
+
+/**
+ * @fileoverview: Handlers for Neo4j driver instrumentation
+ */
+
+const uuid4 = require('uuid4');
+
+const moduleUtils = require('./module_utils.js');
+const eventInterface = require('../event.js');
+const utils = require('../utils.js');
+const serverlessEvent = require('../proto/event_pb.js');
+const tracer = require('../tracer.js');
+const errorCode = require('../proto/error_code_pb.js');
+
+/**
+ * Extracts the relevant arguments from provided arguments
+ * @returns {Object} Object of the extracted arguments
+ */
+function getArgsFromFunction(...args) {
+    return {
+        query: args[0],
+        params: args[1],
+    };
+}
+
+
+/**
+ * Gets session's address info from Transaction/Session instance
+ * @param {neo4j.Session|neo4j.Transaction} session A DB session (could be Session of Transaction)
+ * @returns {Object} json with address info { host, port }
+ */
+function getAddressInfo(session) {
+    let port = '7474';
+    let host = 'neo4j';
+    let connectionHolder;
+
+    switch (session.constructor.name) {
+    // By default, Session opens two same connectionProviders - Read and Write, and not default one
+    case 'Session':
+        connectionHolder = session._readConnectionHolder;
+        break;
+
+    case 'Transaction':
+        connectionHolder = session._connectionHolder;
+        break;
+
+    default:
+        return {
+            host, port,
+        };
+    }
+
+    if (session && connectionHolder && connectionHolder._connectionProvider) {
+        const connectionProvider = connectionHolder._connectionProvider;
+        port = connectionProvider._address.port();
+        host = connectionProvider._address.host();
+    }
+
+    return {
+        port, host,
+    };
+}
+
+
+/**
+ * Parse from DB session with Neo4j server, useful metadata
+ * @param {neo4j.Session|neo4j.Transaction} session A DB session (could be Session of Transaction)
+ * @returns {{metadata: Object, fullMetadata: Object}} JSONs with command's metadata and command's
+ *      full metadata which contains additional data about the event
+ */
+function getNeo4JSessionMetadata(session) {
+    const metadata = {};
+    const fullMetadata = {};
+
+    let connectionHolder;
```

Comment:
Definition is not enough it will raise an exception if you don't init this value in case of default

Reply:
Fixed by added later this function 'if(connectionHandler)'
Is that the right way?
Otherwise i need to init a new connectionHandler instance, right?

Reply:
yea its ok